link-move/src/main/java/com/nhl/link/move/runtime/task/deleteall/DeleteAllTask.java

```diff
@@ -0,0 +1,169 @@
+    private final boolean skipExecutionStats;
+    private final ITargetCayenneService targetCayenneService;
+
+    public DeleteAllTask(
+            Class<?> type,
+            Expression targetFilter,
+            ITargetCayenneService targetCayenneService,
+            ITokenManager tokenManager,
+            DbEntity dbEntity,
+            boolean skipExecutionStats,
+            LmLogger logger) {
+
+        //"deleteAll" task doesn't use extractor and doesn't support batches
+        super(null, 0, tokenManager, logger);
+
+        this.type = type;
+        this.targetFilter = targetFilter;
+        this.targetCayenneService = targetCayenneService;
+        this.dbEntity = dbEntity;
+        this.skipExecutionStats = skipExecutionStats;
+    }
+
+    @Override
+    protected String createLabel() {
+        return "deleteAll";
+    }
+
+    @Override
+    protected void doRun(Execution exec) {
+        ObjectContext context = targetCayenneService.newContext();
+
+        if (targetFilter != null) {
+            deleteWithExpression(context, exec);
+        } else {
+            deleteAllData(context, exec);
+        }
+
+        context.commitChanges();
+    }
+
+    @Override
+    protected void onExecFinished(Execution exec) {
+        exec.getLogger().deleteExecFinished();
+    }
+
+    private void deleteWithExpression(ObjectContext context, Execution execution) {
+        StringBuilder queryBuilder = new StringBuilder("DELETE FROM " + type.getName() + " e WHERE ");
+
+        try {
+            targetFilter.appendAsEJBQL(queryBuilder, "e");
+        } catch (IOException e) {
+            throw new LmRuntimeException("Error while applying target filter expression", e);
+        }
+
+        EJBQLQuery delete = new EJBQLQuery(queryBuilder.toString());
+
+        QueryResponse result = context.performGenericQuery(delete);
+
+        incrementTaskResult(result.firstUpdateCount()[0], execution.getStats());
+    }
+
+    private void deleteAllData(ObjectContext context, Execution execution) {
+        DbAdapter adapter = resolveNodeName()
+                .flatMap(targetCayenneService::dbAdapter)
+                .orElseThrow(() -> new LmRuntimeException("Adapter not found for dataMap: " + dbEntity.getDataMap().getName()));
+
+        SQLTemplate query = new SQLTemplate(dbEntity.getDataMap(), "DELETE FROM $tableName", false);
+        query.setParams(Map.of("tableName", adapter.getQuotingStrategy().quotedFullyQualifiedName(dbEntity)));
+
+        TRUNCATE_AWARE_ADAPTERS.forEach(a -> query.setTemplate(a.getName(), "TRUNCATE TABLE $tableName"));
+
+        Optional<Long> prefetchedOverallCount = prefetchCountIfRequired(adapter.unwrap(), context);
+
+        QueryResponse result = context.performGenericQuery(query);
+
+        long deletedCount = prefetchedOverallCount.orElse((long) result.firstUpdateCount()[0]);
+        incrementTaskResult(deletedCount, execution.getStats());
+    }
+
+    private Optional<String> resolveNodeName() {
```

Comment:
Let me think, maybe there's a more straightforward adapter lookup logic. But the approach here is totally acceptable.